neptunecontrib/viz/parallel_coordinates_plot.py

```diff
@@ -0,0 +1,129 @@
+        min_running_time (:obj:`int`, optional, default is ``None``):
+            Minimum running time of an experiment in seconds, like ``2000``.
+
+    Examples:
+
+        .. code:: python3
+
+            # make visualization for all experiments in project
+            make_parallel_coordinates_plot()
+
+            # make visualization for experiment with tag 'segmentation' and save to html file.
+            make_parallel_coordinates_plot(html_file_path='visualizations.html', tag='segmentation')
+
+            # make visualization for all experiments created by john and use selected columns
+            make_parallel_coordinates_plot(
+                columns=['channel_epoch_acc', 'channel_epoch_loss',
+                         'parameter_lr', 'parameter_dense', 'parameter_dropout'],
+                owner='john')
+    """
+    if neptune.project is None:
+        msg = """You do not have project, from which to fetch data.
+                 Use neptune.init() to set project, for example: neptune.init('USERNAME/example-project').
+                 See docs: https://docs.neptune.ai/neptune-client/docs/neptune.html#neptune.init"""
+        raise ValueError(msg)
+
+    df = neptune.project.get_leaderboard(id=id, state=state, owner=owner, tag=tag, min_running_time=min_running_time)
+    assert df.shape[0] != 0, 'No experiments to show. Try other filters.'
+
+    if columns is None:
+        columns = ['id', 'owner']
+        for col_name in df.columns.to_list():
+            if 'parameter_' in col_name:
+                columns.append(col_name)
+    elif isinstance(columns, str):
+        assert columns in df.columns.to_list(), f'There is no "{columns}" in the project columns.'
+        columns = ['id', columns]
+    elif isinstance(columns, list):
+        if 'id' not in columns:
+            columns.append('id')
+        assert all(column in df.columns.to_list() for column in columns), \
+            'Check "columns" parameter, for columns that are not in this project.'
+    else:
+        raise TypeError(f'{columns} must be None, string or list of string')
+
+    # sort experiments by neptune id
+    df = df[columns]
+    df = df.rename(columns={'id': 'neptune_id'})
+    _exp_ids_series = df['neptune_id'].apply(lambda x: int(x.split('-')[-1]))
+    df.insert(loc=0, column='exp_number', value=_exp_ids_series)
+    df = df.sort_values(by='exp_number', ascending=True)
+    df = df.astype({'neptune_id': 'string'})
+
+    # prepare HiPlot visualization
+    input_to_hiplot = df.T.to_dict().values()
+    hiplot_vis = hip.Experiment().from_iterable(input_to_hiplot)
+    for j, datapoint in enumerate(hiplot_vis.datapoints[1:], 1):
+        datapoint.from_uid = hiplot_vis.datapoints[j-1].uid
+
+    # save to html if requested
+    if html_file_path is not None:
+        assert isinstance(html_file_path, str),\
+            f'"html_file_path" should be string, but {type(html_file_path)} is given'
+        if len(os.path.dirname(html_file_path)) > 0:
```

Comment:
Suggested way of checking if list is empty is to go:

```python
if my_list:
   pass
```

Travis job fails because it spotted that you are doing if `len(my_list) > 0`

Reply:
I'll correct to `if os.path.dirname(html_file_path):` thanks